pkg/kademlia/queue.go

```diff
@@ -4,57 +4,99 @@
 package kademlia
 
 import (
+	"container/heap"
 	"math/big"
 
+	"storj.io/storj/pkg/dht"
 	"storj.io/storj/pkg/pb"
 )
 
-// An Item is something we manage in a priority queue.
-type Item struct {
+//XorQueue is a priority queue where the priority is key XOR distance
+type XorQueue struct {
+	maxLen int
+	items  items
+}
+
+//NewXorQueue returns a items with priority based on XOR from targetBytes
+func NewXorQueue(size int) *XorQueue {
+	return &XorQueue{items: make(items, 0, size), maxLen: size}
```

Comment:
this is limiting the queue to a maximum size but we want to let it grow larger right?

Reply:
I think we use append to grow items later, which should safely resize the array.  see example https://play.golang.org/p/IBCJSqJaSf0.  All this really does is preallocate memory.

Reply:
perfect I thought this limited the size it could grow to. Get my languages confused sometimes

Reply:
yeah, I had to double check!